src/wp-admin/network/settings.php

```diff
@@ -254,14 +254,16 @@
 				<td>
 					<?php
 					$limited_email_domains = get_site_option( 'limited_email_domains' );
-					$limited_email_domains = str_replace( ' ', "\n", $limited_email_domains );
 
-					if ( $limited_email_domains ) {
-						$limited_email_domains = implode( "\n", (array) $limited_email_domains );
+					$limited_email_domains_string = '';
+					if ( is_array( $limited_email_domains ) ) {
+						$limited_email_domains_string = implode( "\n", $limited_email_domains );
+					} elseif ( is_string( $limited_email_domains ) ) {
+						$limited_email_domains_string = str_replace( ' ', "\n", $limited_email_domains );
```

Comment:
For <= PHP 8.0. no changes are needed. For 8.1, null will generate a deprecation. Can check for null and set to empty string else do the `str_replace`.

The `implode` is not necessary when not an array. For better quality and stability, check if an array and if yes, then do the `implode`.

https://gist.github.com/jrfnl/106ff1b0585f88862e5bb030a332be72